rtls/monitor.go

```diff
@@ -0,0 +1,83 @@
+package rtls
+
+import (
+	"crypto/tls"
+	"log"
+	"os"
+	"sync/atomic"
+	"time"
+)
+
+// monitorFiles watches the given files for changes, calling the supplied callback when a
+// change is detected. The callback is passed a slice of the files that changed. The loop
+// never terminates, so it's expected the caller will invoke this in a goroutine.
+func monitorFiles(files []string, cb func([]string)) {
+	mtimes := make([]time.Time, len(files))
+	for {
+		var changed []string
+		for n, file := range files {
+			info, err := os.Stat(file)
+			if err != nil {
+				continue
+			}
+			if !mtimes[n].IsZero() && info.ModTime() != mtimes[n] {
+				changed = append(changed, file)
+			}
+			mtimes[n] = info.ModTime()
+		}
+		if changed != nil {
+			cb(changed)
+		}
+		time.Sleep(2 * time.Second)
```

Comment:
This is not idiomatic Go. You should use the ticker pattern.

https://gobyexample.com/tickers

Reply:
Sure, I can adapt it easily enough. It's not clear to me what value using a ticker and channel will provide over callback in this case.  A ticker makes sense when you want to select on multiple things (most notably a context), but in this case it was literally just a "do a thing every interval T."

> Also, this could could be quite racy, since I don't see any read-write synchronization being done.

It uses an atomic.Value to hold a scalar (a pointer to tls.Certificate).

Reply:
>  I would expect to see use of the "done" channel pattern.

What is it that will signal doneness?

In this case there's no cleanup activity for the goroutine to do, and it doesn't impede rqlited's ability to terminate.

Reply:
To be clear, I'm not objecting to making the changes, just asking to be administered the cluebat a little bit more. :)